src/main/java/com/voice/dbRegistration/restController/DevoteeInfoRestController.java

```diff
@@ -43,15 +52,16 @@ private DevoteeInfo encryptData(DevoteeInfo input) {
     }
 
     @PostMapping("/doesUserExist")
-    public List<DevoteeInfo> doesExist(@RequestBody Map<String, String> input) {
-        String email = input.get("email");
-        return devoteeInfoDao.findAllByEmail(email);
+    public ResponseEntity<List<DevoteeInfo>> doesExist(Authentication authentication) {
+        Optional<UserAuth> user=userAuthService.getUserAuthFromAuthentication(authentication);
+        return user.map(userAuth -> ResponseEntity.ok(devoteeInfoDao.findAllByEmail(userAuth.getUserEmail()))).orElseGet(() -> ResponseEntity.status(HttpStatus.UNAUTHORIZED).build());
+
```

Comment:
We should get only one user from db query. So why are we returning List<DevoteeInfo>?

Reply:
applied changes